qiskit/dagcircuit/dagcircuit.py

```diff
@@ -610,14 +610,20 @@ def compose(self, other, edge_map=None, qubits=None, clbits=None, front=False, i
             return None
 
     def idle_wires(self):
-        """Return idle wires.
+        """Return idle wires (wires which only contain barriers are idle).
```

Comment:
Since this change the default behavior of `idle_wires`, I would add a parameter like `barrier_is_idle` with default `False` to preserver the previous behavior. The drawer can call this method with an explicit `barrier_is_idle=True`.